packages/manager/modules/pci/src/projects/project/storages/databases/database/pools/pools.controller.js

```diff
@@ -0,0 +1,42 @@
+export default class {
+  /* @ngInject */
+  constructor($translate, CucCloudMessage) {
+    this.$translate = $translate;
+    this.CucCloudMessage = CucCloudMessage;
+  }
+
+  $onInit() {
+    this.initData();
+    this.ADD_POOL_TRACKING = `${this.databasesTrackPrefix}::pools::add_pool_cancel`;
+    this.loadMessages();
+    this.trackDashboard('pools', 'page');
+    this.messageContainer =
+      'pci.projects.project.storages.databases.dashboard.pools';
+  }
+
+  /**
+   * Function to map and add database name and user name to pool class
+   */
+  initData() {
+    this.pools.forEach((pool) => {
+      pool.setDatabaseName(
+        this.postgresDatabases.find((db) => db.id === pool.databaseId),
+      );
+      pool.setUserName(this.users.find((user) => user.id === pool.userId));
+    });
+  }
```

Comment:
**suggestion:** let's avoid looping inside a loop
```
let databases = {};
this.postgresDatabases.forEach((db) => databases[db.id] = db);
pool.setDatabaseName(databases[pooldatabaseId);
```
Same could also be used for setting username.

Reply:
Databases is an array here not object...if i want to create an object structure, i need to restructure it which will involve looping it again. I dont see how it could improve performance ?

Reply:
@varun257 Yes.

Reply:
Oh okay but keeping in mind that these arrays are required in next modal as well to create pool. I think its better to keep it as array